src/event/validation.ts

```diff
@@ -36,7 +36,10 @@ export const isDefined = (v: unknown): boolean => v !== null && typeof v !== "un
 export const isBoolean = (v: unknown): boolean => typeof v === "boolean";
 export const isInteger = (v: unknown): boolean => Number.isInteger(v as number);
 export const isDate = (v: unknown): v is Date => v instanceof Date;
-export const isBinary = (v: unknown): v is Uint32Array => v instanceof Uint32Array;
+export const isBinary = (v: unknown): boolean =>
+  v instanceof Uint32Array ||
+  v instanceof Uint16Array ||
+  v instanceof Uint8Array;
```

Comment:
I think there must be a better way to do this than to hardcode all typed arrays.

What about `ArrayBuffer.isView`?